### ISSUE TITLE
change setLabelFormat to public

### ISSUE THREAD

**Issue by hikashop-nicolas** [myparcelnl/sdk#149]
I am generating the shipment ids with createConcepts and then retrieving them with getConsignmentsByReferenceId and getMyParcelConsignmentId
Then, I'm storing the shipment ids with the order data in the database.
Later on, I want to allow the merchant to print the label of several orders using the shipment id stored in each order data (to avoid creating a new concept).
So I have this function:

`
	private function _getLabel($shipment_ids) {
		$config = hikashop_config();
		$myParcelCollection = (new MyParcelCollection())
			->setUserAgent('HikaShop', $config->get('version'));
		if(!is_array($shipment_ids))
			$shipment_ids = array($shipment_ids);
		// NOTE: We had to change that function to be public in the SDK
		$myParcelCollection->setLabelFormat($this->plugin_options['positions']);
		$request = (new MyParcelRequest())
			->setUserAgent($myParcelCollection->getUserAgent())
			->setRequestParameters(
				$this->plugin_options['api_key'],
				implode(';', $shipment_ids) . '/' . $myParcelCollection->getRequestBody(),
				MyParcelRequest::REQUEST_HEADER_RETRIEVE_LABEL_LINK
			)
			->sendRequest('GET', MyParcelRequest::REQUEST_TYPE_RETRIEVE_LABEL);
		return MyParcelRequest::REQUEST_URL . $request->getResult('data.pdfs.url');
	}
`

But I had to change setLabelFormat to public in the SDK as I couldn't find any alternative to provide the shipment ids myself to the SDK to generate the label in the format I want.
Could this be changed ? I don't see any drawback in making that function public.
Or maybe a function could be added to get a label generated from an array of shipment ids ?

**Comment by hikashop-nicolas** [comment]
Sorry, it seems github is messing with the indenting and line returns in code when you edit it...

`
	private function _getLabel($shipment_ids) {
		$config = hikashop_config();
		$myParcelCollection = (new MyParcelCollection())
			->setUserAgent('HikaShop', $config->get('version'));

		if(!is_array($shipment_ids))
			$shipment_ids = array($shipment_ids);

		// NOTE: We had to change that function to be public in the SDK
		$myParcelCollection->setLabelFormat($this->plugin_options['positions']);
		$request = (new MyParcelRequest())
			->setUserAgent($myParcelCollection->getUserAgent())
			->setRequestParameters(
				$this->plugin_options['api_key'],
				implode(';', $shipment_ids) . '/' . $myParcelCollection->getRequestBody(),
				MyParcelRequest::REQUEST_HEADER_RETRIEVE_LABEL_LINK
			)
			->sendRequest('GET', MyParcelRequest::REQUEST_TYPE_RETRIEVE_LABEL);
		return MyParcelRequest::REQUEST_URL . $request->getResult('data.pdfs.url');
	}
`

**Comment by hikashop-nicolas** [comment]
It seems github is messing the formatting of the code. Here is a screenshot of the code instead:
![Web - hikashop_plugins_plg_hikashop_myparcel_myparcel php - Aptana Studio 3 2019-07-29 16 04 54](https://user-images.githubusercontent.com/6094338/62054826-a80f9c80-b21a-11e9-8cbb-b86cade4ca95.png)

**Comment by Mark-H** [comment]
I think that can be done with `setLinkOfLabels` or `setPdfOfLabels`. Their first parameter is the label format/position (1-4 = position on a4, false = a6). Internally that calls the setLabelFormat method you're looking for. 

While those methods also call createConcepts(), from reading the code it seems it should only do that for consigments where the myparcel_consignment_id is null, i.e. actual new ones.  

I'm using it like this for new :

```
        $collection = (new MyParcelCollection())
            ->setUserAgent('modmore\Commerce', (string)$this->commerce->version)
            ->addConsignment($consignment)
            ->setLinkOfLabels(1);
```

In your case I think you should be able of using the SDK like this to fetch the right consigments and to generate the labels:

```
$collection = MyParcelCollection::findMany($shipment_ids, $this->plugin_options['api_key']);
$link = $collection->setLinkOfLabels($this->plugin_options['positions'])->getLinkOfLabels();
// or
$collection->setPdfOfLabels($this->plugin_options['positions'])->downloadPdfOfLabels();
```

(To not have github mess with the code formatting, wrap it in triple backticks before and after the code block, on their own line.)

**Comment by RichardPerdaan** [comment]
I see no reason why we could not set the function `setLabelFormat` to public, so I also opened the following pull request: https://github.com/myparcelnl/sdk/pull/154
When it is approved by my colleagues I will process it in a new version.

**Comment by RichardPerdaan** [comment]
The new version is live, the change has been included in the version [2.1.4](https://github.com/myparcelnl/sdk/releases/tag/v2.1.4)
